### PR TITLE
Testcase for gcc package,this testcase make use of testsuite provided…

### PR DESCRIPTION
**Pull request by ramyabs**
… by the source package.

Description :
The gcc package contains the GNU Compiler Collection version.
You'll need this package in order to compile C code.

Signed-off-by: Ramya BS ramya@linux.vnet.ibm.com
Tested-by     : Ramya BS ramya@linux.vnet.ibm.com

### CODE REVIEW

File: toolchain/gcc.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                             'makeinfo', 'texinfo', 'mpc-devel'])
+        else:
+            packages.extend(['glibc-static', 'autogen', 'guile',
+                             'guile-devel', 'libgo', 'libgo-devel',
+                             'libgo-static', 'elfutils-devel',
+                             'texinfo-tex', 'texinfo', 'elfutils-libelf-devel',
+                             'gmp-devel', 'mpfr-devel', 'libmpc-devel',
+                             'gcc-gnat', 'libgnat', 'zlib-devel',
+                             'gettext', 'sharutils', 'libgcj', 'libunwind'
+                             'libgcc', 'libgomp'])
```
Comment (PraveenPenguin) [inline toolchain/gcc.py]: i think u already added `sharutils` in global packages can u please remove this

Comment (PraveenPenguin) [inline toolchain/gcc.py]: for Fedora 24 i think libgcj is not available can you please help me on this 
```

[root@localhost ~]# yum install libgcj
Yum command has been deprecated, redirecting to '/usr/bin/dnf install libgcj'.
See 'man dnf' and 'man yum2dnf' for more information.
To transfer transaction metadata from yum to DNF, run:
'dnf install python-dnf-plugins-extras-migrate && dnf-2 migrate'

Last metadata expiration check: 0:11:27 ago on Thu Oct 13 14:41:20 2016.
No package libgcj available.
Error: Unable to find a match.
[root@localhost ~]#
```

File: toolchain/gcc.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                'gcc \(SUSE Linux\) (.*)\n',
+                gcc_version).group(1).replace(
+                ".",
+                "_")
+        else:
+            get_system_gcc_version = process.system_output(
+                "gcc -dumpversion").replace(".", "_").split("\n", 1)[0]
+        tarball = self.fetch_asset(
+            "https://github.com/gcc-mirror/gcc/archive/gcc-%s-release.zip" %
+            get_system_gcc_version, expire='7d')
```
Comment (PraveenPenguin) [inline toolchain/gcc.py]: in fedora 24 it fails

```

Traceback (most recent call last):
  File "/root/gcc.py", line 78, in setUp
    get_system_gcc_version, expire='7d')
  File "/usr/lib/python2.7/site-packages/avocado_framework-41.0-py2.7.egg/avocado/core/test.py", line 654, in fetch_asset
    self.cache_dirs, expire).fetch()
  File "/usr/lib/python2.7/site-packages/avocado_framework-41.0-py2.7.egg/avocado/utils/asset.py", line 142, in fetch
    raise EnvironmentError("Failed to fetch %s." % self.basename)
EnvironmentError: Failed to fetch gcc-6_2_1-release.zip.

ERROR 1-gcc.py:GCC.test -> TestSetupFail: Failed to fetch gcc-6_2_1-release.zip.

Not logging /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor (file does not exist)
Not logging /proc/pci (file does not exist)
Test results availab 


```
this is gcc version 
[root@localhost ~]# gcc -dumpversion
6.2.1
[root@localhost ~]#

Comment (PraveenPenguin) [inline toolchain/gcc.py]: @ramyabs can you please use upstream git url that can be common for all if you want to make support fedora/centos 


`https://github.com/gcc-mirror/gcc/archive/master.zip ` use this url

File: toolchain/gcc.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        archive.extract(tarball, self.srcdir)
+        gcc_folder = 'gcc-%s' % os.path.basename(tarball.split('.zip')[0])
+        self.srcdir = os.path.join(self.srcdir, gcc_folder)
+        os.chdir(self.srcdir)
+        process.run('./configure', ignore_status=True, sudo=True)
+        build.make(self.srcdir)
+
```
Comment (PraveenPenguin) [inline toolchain/gcc.py]: please use` ignore_status=True` if command failed script will error out

File: toolchain/gcc.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    def test(self):
+        """
+        Runs the gcc `make check`
+        """
+        ret = build.run_make(self.srcdir, extra_args='check')
```
Comment (PraveenPenguin) [inline toolchain/gcc.py]: here also please 
`ret = build.run_make(self.srcdir, extra_args='check',ignore_status=True)`

File: toolchain/gcc.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        if dist.name == 'SuSE':
+            gcc_version = process.system_output("gcc --version")
+            get_system_gcc_version = re.search(
+                'gcc \(SUSE Linux\) (.*)\n',
+                gcc_version).group(1).replace(
+                ".",
+                "_")
+        else:
+            get_system_gcc_version = process.system_output(
+                "gcc -dumpversion").replace(".", "_").split("\n", 1)[0]
```
Comment (PraveenPenguin) [inline toolchain/gcc.py]: sample code can be for upstream git :

  ```
      locations = ["https://github.com/gcc-mirror/gcc/archive/"
                     "master.zip"]
        tarball = self.fetch_asset("gcc.zip", locations=locations,
                                   expire='7d')
        archive.extract(tarball, self.srcdir)
        self.srcdir = os.path.join(self.srcdir, 'gcc-master')


        os.chdir(self.srcdir)
        process.run('./configure', ignore_status=True, sudo=True)
        build.make(self.srcdir,ignore_status=True)
```

File: toolchain/gcc.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        tarball = self.fetch_asset(
+            "https://github.com/gcc-mirror/gcc/archive/gcc-%s-release.zip" %
+            get_system_gcc_version, expire='7d')
+        archive.extract(tarball, self.srcdir)
+        gcc_folder = 'gcc-%s' % os.path.basename(tarball.split('.zip')[0])
+        self.srcdir = os.path.join(self.srcdir, gcc_folder)
+        os.chdir(self.srcdir)
+        process.run('./configure', ignore_status=True, sudo=True)
```
Comment (narasimhan-v) [inline toolchain/gcc.py]: @PraveenPenguin agreed

**Comment by PraveenPenguin** [comment]
@ramyabs if you still want to pursue this test case (PR) please raise a new PR ,

**Pull request closed without merging**